### ISSUE TITLE
what if all gradients goes to zeros?

### ISSUE THREAD

**Issue by MrWanter** [pytorch/pytorch#9243]
Hi! I'm modifying some code [pytorch-faster-rcnn](https://github.com/jwyang/faster-rcnn.pytorch), I after adding some layers and loss term, in the training, the loss is going down, but after several steps in inital training, all gradients goes to zeros, but loss still not low, I checked when beginning training the gradients are normal, very puzelled about this behavior, can someone give me some general clue of how to debug this kind of problem?

**Comment by soumith** [comment]
please use https://discuss.pytorch.org for questions
